core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -694,6 +697,16 @@ public ServerBuilder meterRegistry(MeterRegistry meterRegistry) {
         return this;
     }
 
+    /**
+     * Sets a global naming rule for the name of services.
```

Comment:
How about adding per-route service naming as well? We could add `AbstractServiceBindingBuilder.serviceNaming(..)` whose default is `null`. Then, we could re-implement `AbstractServiceBindingBuilder.defaultServiceName()` as a shortcut to `serviceNaming(cfg -> defaultServiceName)` or `serviceNaming(ServiceNaming.of("myService"))`.

We should also add `VirtualHostBuilder.serviceNaming(..)` whose default is `null` as well.

Please see `VirtualHostBuilder.build(VirtualHostBuilder template)` to understand how we derive each `ServiceConfig`'s properties from `VirtualHostBuilder` and `ServerBuilder`. It's a little bit complicated, so please feel free to let us know if you have any questions or need any help.

Reply:
Thanks for guidances and kindness. Actually I was going to update `ServiceBuindingBinder` instead of `ServerBuilder` at first. https://github.com/line/armeria/issues/3232#issuecomment-775790003 I will take a look again!